### ISSUE TITLE
[bug] `@unocss/transformer-variant-group` not working with HMR with `@unocss/nuxt`

### ISSUE THREAD

**Issue by itsmnthn** [unocss/unocss#817]
I'm using `@unocss/nuxt` in `nuxt3` with `vite` on `node v17.6.0`

I'm using `@unocss/transformer-variant-group` to use group variants like this

```HTML
<div class="mx-auto rounded-2xl grb-b before:(bg-an-gradient-rainbow-190 p-0.5 grb-gray-raisin bg-w-250 bg-xy-100-0) hover:(before:bg-xy-0-0 transition-all duration-300)">
  <div class="p-10">
    {{ $t('hello') }}
     <br />
    {{ $t('welcome', { name: 'world' }) }}
  </div>
</div>
```

as expected styles are being transformed to on reload or initially
```HTML
<div class="mx-auto rounded-2xl grb-b before:bg-an-gradient-rainbow-190 before:p-0.5 before:grb-gray-raisin before:bg-w-250 before:bg-xy-100-0 hover:before:bg-xy-0-0 hover:transition-all hover:duration-300" >
</div>
```
![Screenshot 2022-04-11 at 5 08 15 PM](https://user-images.githubusercontent.com/25129674/162731911-9b8c3465-c36c-46eb-8b5f-518d3385638e.png)

after making changes to the file, transformation doesn't work on HMR it pushes classes as written 
```HTML
<div class="mx-auto rounded-2xl grb-b before:(bg-an-gradient-rainbow-190 p-0.5 grb-gray-raisin bg-w-250 bg-xy-100-0) hover:(before:bg-xy-0-0 transition-all duration-300)">
</div>
```
![image](https://user-images.githubusercontent.com/25129674/162732105-246c36ac-4af1-4afe-a6b0-5b7729561ddd.png)

**Comment by itsmnthn** [comment]
Getting `Failed to parse bg-xy-100-0) Error: Unbalanced parenthesis` error on output log on vscode
![image](https://user-images.githubusercontent.com/25129674/162732692-83f1f889-8d5b-4fe1-a008-ba2e7986c6f4.png)

**Comment by qin-guan** [comment]
I think this might be linked to https://github.com/unocss/unocss/issues/954. I am also experiencing this issue.

Here's a repro: https://stackblitz.com/edit/github-qz67ei?file=pages/index.vue

**Comment by jd-solanki** [comment]
Hi all 👋🏻 

I have solved mine #954 by adding `configDeps` option in UnoCSS config

```ts
configDeps: ['../anu-vue/src/presets/theme-default/index.ts'],
```

`index.ts` is a single preset file where I make changes and it resolved my issue.

Please let me know if works for you.

**Comment by itsmnthn** [comment]
I think you are right this happens for the page only. Components Used on page works as expected.
